torchvision/ops/dropblock.py

```diff
@@ -1 +1,64 @@
 import torch
+import torch.nn.functional as F
+from torch import nn, Tensor
+
+from ..utils import _log_api_usage_once
+
+
+class DropBlock2d(nn.Module):
+    """
+    Implements DropBlock2d from `"DropBlock: A regularization method for convolutional networks"
+    <https://arxiv.org/abs/1810.12890>`.
+
+    Args:
+        p (float): Probability of an element to be dropped.
+        block_size (int): Size of the block to drop.
+        inplace (bool): If set to ``True``, will do this operation in-place. Default: ``False``
+    """
+
+    def __init__(self, p: float, block_size: int, inplace: bool = False) -> None:
+        super(DropBlock2d, self).__init__()
+        _log_api_usage_once(self)
+
+        if p < 0.0 or p > 1.0:
+            raise ValueError(f"drop probability has to be between 0 and 1, but got {p}")
+        self.p = p
+        self.block_size = block_size
+        self.inplace = inplace
+
+    def forward(self, input: Tensor) -> Tensor:
+        """
+        Args:
+            input (Tensor): Input feature map on which some areas will be randomly
+                dropped.
+        Returns:
+            Tensor: The tensor after DropBlock layer.
+        """
+        if not self.training:
```

Comment:
Skip early if p is zero:
```suggestion
        if not self.training or self.p == 0.0:
```